src/common/device/AndroidDevice.ts

```diff
@@ -5,22 +5,33 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
 
+import * as childProcess from 'node:child_process';
 import fs from 'node:fs';
 import path from 'node:path';
 import os from 'node:os';
-import { Logger, SfError } from '@salesforce/core';
+import { Logger, Messages, SfError } from '@salesforce/core';
 import { AndroidUtils } from '../AndroidUtils.js';
 import { Version } from '../Common.js';
 import { CryptoUtils, SSLCertificateData } from '../CryptoUtils.js';
+import { CommonUtils } from '../CommonUtils.js';
 import { BaseDevice, DeviceType, LaunchArgument } from './BaseDevice.js';
 
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/lwc-dev-mobile-core', 'common');
+
 export enum AndroidOSType {
-    googleAPIs = 'google_apis',
-    googlePlayStore = 'google_apis_playstore',
-    androidDesktop = 'android-desktop',
-    googleTV = 'google-tv',
-    androidWear = 'android-wear',
-    androidAutomotive = 'android-automotive'
+    googleAPIs = 'google apis',
+    googlePlayStore = 'google apis playstore',
+    androidDesktop = 'android desktop',
+    googleTV = 'google tv',
+    androidWear = 'android wear',
+    androidAutomotive = 'android automotive'
+}
+
+export enum BootMode {
+    normal = 'normal',
+    systemWritablePreferred = 'systemWritablePreferred',
```

Comment:
Where would we end up using "preferred" vs. "mandatory", out of curiosity?

Reply:
For local dev preview on mobile, the first step would be to boot the device, then see if it has cert installed or not, and if not then install the cert (but to install the cert we need the device to be booted with write system access).

So the LDP CLI command first attempts to boot a device with `preferred` writable-system. If the device is not already booted then we boot it with writable system right off the bat assuming that there may be a need to install the cert and in order to avoid a forced reboot into writable system later one, in order to save time. Even if the device already has the cert installed, booting it in writable system doesn't cause any issues with local dev previewing.

If the device is already booted (for example by AS) then we just continue and won't force-reboot into writable-system. Then we check to see if cert is installed or not. If it is installed then all is good and we continue to use the already booted device as is. But if it is not installed then we force-reboot into writable system.

So having this "preferred" vs. "mandatory" can help us get to local preview faster under certain conditions (b/c of less device reboot)